tools/btc_proof_generator/src/main.rs

```diff
@@ -0,0 +1,143 @@
+use anyhow::Result;
+use bitcoin::{
+    consensus::{deserialize, encode::serialize_hex},
+    Block,
+};
+use bitcoin_spv::{
+    btcspv::hash256_merkle_step,
+    types::{Hash256Digest, MerkleArray},
+    validatespv,
+};
+use clap::{load_yaml, value_t, App};
+use hex::FromHex;
+use serde::{Deserialize, Serialize};
+
+fn get_merkle_proof(block: &Block, index: usize) -> Result<Vec<Vec<u8>>> {
+    let tx_len = block.txdata.len();
+    assert!(tx_len >= 1);
+    if tx_len == 1 {
+        return Ok(vec![block.txdata[0].txid().to_vec()]);
+    }
+    let mut layers = Vec::new();
+    let mut proof = Vec::new();
+    let layer: Vec<_> = block
+        .txdata
+        .iter()
+        .map(|tx| tx.txid().as_ref().to_vec())
+        .collect();
+    layers.push(layer);
+    let mut current_layer_index = 0;
+    let mut proof_index_in_layer = index;
+    loop {
+        let mut current_layer = layers[current_layer_index].iter();
+        let current_layer_len = current_layer.len();
+        if current_layer_len <= 1 {
+            break;
+        }
+        let new_layer_len = current_layer_len / 2 + current_layer_len % 2;
+        let mut new_layer = Vec::with_capacity(new_layer_len);
+        while let Some(hash1) = current_layer.next() {
+            let hash2 = current_layer.next().unwrap_or(hash1);
+            new_layer.push(hash256_merkle_step(hash1, hash2).as_ref().to_vec());
+        }
+        layers.push(new_layer);
+        let current_layer = layers[current_layer_index].clone();
+        if proof_index_in_layer % 2 == 0 {
+            proof.push(current_layer[proof_index_in_layer + 1].clone());
+        } else {
+            proof.push(current_layer[proof_index_in_layer - 1].clone());
+        }
+        proof_index_in_layer >>= 1;
+        current_layer_index += 1;
+    }
+    Ok(proof)
+}
+
+fn fetch_block(block_hash: &str) -> Result<Block> {
+    let url = format!("https://blockchain.info/block/{}?format=hex", block_hash);
```

Comment:
is this get block from main-net ? can we make it a option to user, main-net or test-net

Reply:
It is easy to add the feature. I think we can leave it so until we really need it.